app/services/stripe_service/report.rb

```diff
@@ -0,0 +1,159 @@
+class StripeService::Report
+
+  private
+
+  attr_reader :tx, :exception
+
+  public
+
+  def initialize(tx:, exception: nil)
+    @tx = tx
+    @exception = exception
+  end
+
+  def capture_charge_start
+    result = capture_charge.merge({
+      "event": "stripe_call",
+    })
+    logger.info('capture_charge_start', nil, result)
+    result
+  end
+
+  def capture_charge_success
+    result = capture_charge.merge({
+      "event": "stripe_call_succeeded",
+      "stripe_charge_state": "success"
+    })
+    logger.info('capture_charge_success', nil, result)
+    result
+  end
+
+  def capture_charge_failed
+    result = capture_charge.merge({
+      "event": "stripe_call_failed",
+      "stripe_error": stripe_error
+    })
+    logger.info('capture_charge_failed', nil, result)
+    result
+  end
+
+  def create_charge_start
+    result = create_charge.merge({
+      "event": "stripe_call",
+    })
+    logger.info('create_charge_start', nil, result)
+    result
+  end
+
+  def create_charge_success
+    result = create_charge.merge({
+      "event": "stripe_call_succeeded",
+      "stripe_payment_id": stripe_payment.id,
+      "stripe_charge_id": stripe_payment.stripe_charge_id,
+      "stripe_charge_state": "success"
+    })
+    logger.info('create_charge_success', nil, result)
+    result
+  end
+
+  def create_charge_failed
+    result = create_charge.merge({
+      "event": "stripe_call_failed",
+      "stripe_payment_id": stripe_payment&.id,
+      "stripe_error": stripe_error
+    })
+    logger.info('create_charge_failed', nil, result)
+    result
+  end
+
+  def create_payout_start
+    result = create_payout.merge({
+      "event": "stripe_call",
+    })
+    logger.info('create_payout_start', nil, result)
+    result
+  end
+
+  def create_payout_success
+    result = create_payout.merge({
+      "event": "stripe_call_succeeded",
+      "stripe_payment_id": stripe_payment.id,
+      "stripe_charge_id": stripe_payment.stripe_charge_id,
+      "stripe_transfer_id": stripe_payment.stripe_transfer_id
```

Comment:
@aivils for some reason this was logged as `nil`, even though the payment does have the column updated. Any idea why?

Reply:
Right, need for `.reload`... Thanks for fixing @aivils 👍